autoload/go/lint_test.vim

```diff
@@ -74,8 +74,10 @@ func! Test_GometaAutoSave() abort
         \ {'lnum': 5, 'bufnr': 2, 'col': 1, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': 'w', 'pattern': '', 'text': 'exported function MissingDoc should have comment or be unexported (golint)'}
       \ ]
 
-  " clear the quickfix lists
-  call setqflist([], 'r')
+  let winnr = winnr()
+
+  " clear the location lists
+  call setloclist(l:winnr, [], 'r')
```

Comment:
Same here. We should have a switch and call `setqflist()` or `setloclist()` based `go#list#Type("GoMetaLinterAutoSave")`

Reply:
This test is only about the autosave case, and it only tests with the default list type `GoMetaLinterAutoSave`. Why would it check the list type before clearing the list?